lib/Conversion/ATenToXTenPass.cpp

```diff
@@ -145,6 +144,53 @@ bool isLongestBranch(OpResult left, OpResult right) {
   return true;
 }
 
+// Tablegen adapter to check if attributes allow a conversion from `ReduceMean`
+// to `GlobalAveragePool`.
+bool isReduceMeanGlobalAveragePool2D(Value dims, Value keepdims) {
+
+  // Get defining operands
+  Torch::PrimListConstructOp dimsOp =
+      llvm::dyn_cast<Torch::PrimListConstructOp>(dims.getDefiningOp());
+  Torch::ConstantBoolOp keepdimsOp =
+      llvm::dyn_cast<Torch::ConstantBoolOp>(keepdims.getDefiningOp());
+
+  if (!dimsOp || !keepdimsOp)
+    return false;
+
+  SmallVector<APInt> axes;
+  for (Value operand : dimsOp.getOperands()) {
+    if (auto axesValue =
+            llvm::dyn_cast<Torch::ConstantIntOp>(operand.getDefiningOp()))
+      axes.push_back(axesValue.value());
+  }
+
+  // Check that arguments match expectations for global average pool
+  if (keepdimsOp.value() == 1 && axes.size() == 2 && axes[0] == 2 &&
+      axes[1] == 3)
+    return true;
+
+  return false;
+}
+
+// Tablegen adapter to check if attributes allow a conversion from `AveragePool`
+// to `GlobalAveragePool`.
+bool isAdaptiveAvgPoolGlobalAveragePool2D(Value outsizes) {
+  Torch::PrimListConstructOp outsizesOp =
+      llvm::dyn_cast<Torch::PrimListConstructOp>(outsizes.getDefiningOp());
+
+  SmallVector<APInt> dims;
+  for (Value operand : outsizesOp.getOperands()) {
+    if (auto axesValue =
+            llvm::dyn_cast<Torch::ConstantIntOp>(operand.getDefiningOp()))
+      dims.push_back(axesValue.value());
+  }
+
+  // Check that arguments match expectations for global average pool
+  if (dims.size() == 2 && dims[0] == 1 && dims[1] == 1)
```

Comment:
Same comment as in `isReduceMeanGlobalAveragePool2D`.

Reply:
The only way to achieve this would be using `int64_t` for instance instead of `APInt`. A direct comparison in llvm is not possible this way.